hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/SpecificPartitionsImapReaderPmsTest.java

```diff
@@ -22,110 +22,233 @@
 import com.hazelcast.map.IMap;
 import com.hazelcast.sql.impl.expression.ConstantExpression;
+import com.hazelcast.sql.impl.expression.Expression;
 import com.hazelcast.sql.impl.type.QueryDataType;
 import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
-import org.junit.Assert;
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
-import static com.hazelcast.jet.TestContextSupport.adaptSupplier;
-import static com.hazelcast.jet.Util.entry;
 import static com.hazelcast.jet.core.Edge.between;
 import static com.hazelcast.jet.core.processor.SinkProcessors.writeMapP;
-import static com.hazelcast.jet.core.test.TestSupport.out;
 import static com.hazelcast.jet.impl.JetServiceBackend.SQL_ARGUMENTS_KEY_NAME;
+import static com.hazelcast.jet.sql.impl.SqlEndToEndTestSupport.JobInvocationObserverImpl;
 import static com.hazelcast.jet.sql.impl.connector.map.SpecificPartitionsImapReaderPms.mapReader;
 import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toSet;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
 
 @RunWith(HazelcastSerialClassRunner.class)
-@Category(QuickTest.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
 public class SpecificPartitionsImapReaderPmsTest extends SimpleTestInClusterSupport {
+    private static final int MEMBERS = 5;
     private static final int ITERATIONS = 1000;
-    private int pKey;
-    private int keyOwnerPartitionId;
+
+    private int coordinatorOwnedPartitionKey;
+    private int coordinatorOwnedPartitionId;
+    // Non-coordinator-owned partitions keys and ids, one per member in cluster.
+    // Coordinator is not included!
+    private int[] perMemberOwnedPKey;
+    private int[] perMemberOwnedPId;
+
     private String mapName;
     private String sinkName;
+
+    private IMap<Integer, Integer> sourceMap;
+    private IMap<Integer, Integer> sinkMap;
+
     private JobConfig jobConfig;
 
+    private JobInvocationObserverImpl jobInvocationObserver;
+
+    Map<Address, int[]> partitionAssignment;
+    Map<Integer, Address> reversedPartitionAssignment;
+
     @BeforeClass
     public static void beforeClass() throws Exception {
-        initialize(5, null);
+        initialize(MEMBERS, null);
     }
 
     @Before
     public void before() throws Exception {
+        partitionAssignment = getPartitionAssignment(instance());
+        reversedPartitionAssignment = new HashMap<>();
+        for (Map.Entry<Address, int[]> entry : partitionAssignment.entrySet()) {
```

Comment:
should be easier using Java streams

Reply:
Taste of choice, IMO.

Reply:
with some static imports:
```java
        reversedPartitionAssignment = partitionAssignment.entrySet().stream()
                .flatMap(entry -> stream(entry.getValue()).mapToObj(pId -> entry(pId, entry.getKey())))
                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
```
not that simple as I hoped